### ISSUE TITLE
vm/dart/v8_snapshot_profile_writer_test failing on Linux simarm/simarm64 configurations

### ISSUE THREAD

**Issue by sstrickl** [dart-lang/sdk#40705]
After [3ef5270f](https://github.com/dart-lang/sdk/commit/3ef5270f7173fc5cc6698051ff4d5f23fe186780), vm/dart/v8_snapshot_profile_writer_test was marked as crashing on configurations
```
dartkp-linux-release-simarm
dartkp-linux-release-simarm64
dartkp-no-bare-linux-release-simarm
dartkp-no-bare-linux-release-simarm64
```

An example:
```
--- Command "vm_compile_to_kernel []" (took 05.000081s):
DART_CONFIGURATION=ReleaseSIMARM64 /b/s/w/ir/pkg/vm/tool/gen_kernel --aot --platform=out/ReleaseSIMARM64/vm_platform_strong.dill -o /b/s/w/ir/out/ReleaseSIMARM64/generated_compilations/dartkp/runtime_tests_vm_dart_v8_snapshot_profile_writer_test/out.dill /b/s/w/ir/runtime/tests/vm/dart/v8_snapshot_profile_writer_test.dart --packages=/b/s/w/ir/.packages -Ddart.developer.causal_async_stacks=true

exit code:
0

--- Command "precompiler" (took 05.000237s):
DART_CONFIGURATION=ReleaseSIMARM64 out/ReleaseSIMARM64/gen_snapshot --snapshot-kind=app-aot-elf --elf=/b/s/w/ir/out/ReleaseSIMARM64/generated_compilations/dartkp/runtime_tests_vm_dart_v8_snapshot_profile_writer_test/out.aotsnapshot --ignore-unrecognized-flags --packages=/b/s/w/ir/.packages /b/s/w/ir/out/ReleaseSIMARM64/generated_compilations/dartkp/runtime_tests_vm_dart_v8_snapshot_profile_writer_test/out.dill

exit code:
0

--- Command "remove_kernel_file" (took 27ms):
DART_CONFIGURATION=ReleaseSIMARM64 rm /b/s/w/ir/out/ReleaseSIMARM64/generated_compilations/dartkp/runtime_tests_vm_dart_v8_snapshot_profile_writer_test/out.dill

exit code:
0

--- Command "vm" (took 01:27.000500s):
DART_CONFIGURATION=ReleaseSIMARM64 out/ReleaseSIMARM64/dart_precompiled_runtime --ignore-unrecognized-flags --packages=/b/s/w/ir/.packages /b/s/w/ir/out/ReleaseSIMARM64/generated_compilations/dartkp/runtime_tests_vm_dart_v8_snapshot_profile_writer_test/out.aotsnapshot

exit code:
-6

stdout:
Running ./pkg/vm/tool/gen_kernel --aot --platform out/ReleaseSIMARM64/vm_platform_strong.dill -o /b/s/w/it3luJlU/v8-snapshot-profile-writerHWMQFX/test.dill ./runtime/tests/vm/dart/v8_snapshot_profile_writer_test.dart
Running out/ReleaseSIMARM64/gen_snapshot --snapshot-kind=app-aot-elf --elf=/b/s/w/it3luJlU/v8-snapshot-profile-elf-nonbare-intstripOYPGQZ/test.snap --strip --no-use-bare-instructions --write-v8-snapshot-profile-to=/b/s/w/it3luJlU/v8-snapshot-profile-elf-nonbare-intstripOYPGQZ/profile.heapsnapshot /b/s/w/it3luJlU/v8-snapshot-profile-writerHWMQFX/test.dill
Subcommand stderr:
Warning: Generating ELF library without DWARF debugging information.

stderr:
===== CRASH =====
si_signo=Segmentation fault(11), si_code=1, si_addr=0xf
version=2.8.0-edge.3ef5270f7173fc5cc6698051ff4d5f23fe186780 (Thu Feb 20 01:14:15 2020 +0000) on "linux_simarm64"
thread=30138, isolate=main(0x55ed7d0c5800)
isolate_instructions=7f40f05e1000, vm_instructions=7f40f05dc000
  pc 0x000055ed7b2517b5 fp 0x00007f40ef0fc2e0 dart::Class::IsSubtypeOf(dart::NNBDMode, dart::Class const&, dart::TypeArguments const&, dart::AbstractType const&, dart::Heap::Space)
  pc 0x000055ed7b251649 fp 0x00007f40ef0fc380 dart::Class::IsSubtypeOf(dart::NNBDMode, dart::Class const&, dart::TypeArguments const&, dart::AbstractType const&, dart::Heap::Space)
  pc 0x000055ed7b251488 fp 0x00007f40ef0fc420 dart::Class::IsSubtypeOf(dart::NNBDMode, dart::Class const&, dart::TypeArguments const&, dart::AbstractType const&, dart::Heap::Space)
  pc 0x000055ed7b25d0ce fp 0x00007f40ef0fc4a0 dart::Function::IsSubtypeOf(dart::NNBDMode, dart::Function const&, dart::Heap::Space) const
  pc 0x000055ed7b2d8a39 fp 0x00007f40ef0fc670 dart::DRT_TypeCheck(dart::NativeArguments)
  pc 0x000055ed7b2f3ae4 fp 0x00007f40ef0fc7a0 dart::Simulator::DoRedirectedCall(dart::Instr*)
  pc 0x000055ed7b2f17e5 fp 0x00007f40ef0fc860 dart::Simulator::InstructionDecode(dart::Instr*)
  pc 0x000055ed7b2f946b fp 0x00007f40ef0fc930 dart::Simulator::Execute()
  pc 0x000055ed7b2f9767 fp 0x00007f40ef0fc9e0 dart::Simulator::Call(long, long, long, long, long, bool, bool)
  pc 0x000055ed7b20bdb9 fp 0x00007f40ef0fca90 dart::DartEntry::InvokeFunction(dart::Function const&, dart::Array const&, dart::Array const&, unsigned long)
  pc 0x000055ed7b20ddaf fp 0x00007f40ef0fcae0 dart::DartLibraryCalls::HandleMessage(dart::Object const&, dart::Instance const&)
  pc 0x000055ed7b22cca7 fp 0x00007f40ef0fccd0 dart::IsolateMessageHandler::HandleMessage(std::__2::unique_ptr<dart::Message, std::__2::default_delete<dart::Message> >)
  pc 0x000055ed7b23a370 fp 0x00007f40ef0fcd50 dart::MessageHandler::HandleMessages(dart::MonitorLocker*, bool, bool)
  pc 0x000055ed7b23ad30 fp 0x00007f40ef0fcdb0 dart::MessageHandler::TaskCallback()
  pc 0x000055ed7b30f057 fp 0x00007f40ef0fcde0 dart::ThreadPool::Worker::Loop()
  pc 0x000055ed7b30eede fp 0x00007f40ef0fce10 dart::ThreadPool::Worker::Main(unsigned long)
  pc 0x000055ed7b2a4a46 fp 0x00007f40ef0fced0 out/ReleaseSIMARM64/dart_precompiled_runtime+0x7b1a46
-- End of DumpStackTrace
[exit     : sp(0) fp(0x55ed7d379ce0) pc(0)]
[bare-stub : sp(0x55ed7d379cf0) fp(0x55ed7d379d38) pc(0x7f40f05dca74)]
[bare-dart : sp(0x55ed7d379d48) fp(0x55ed7d379d68) pc(0x7f40f06c1d20) *dart:async__FutureListener@4048458_handleValue ]
[bare-dart : sp(0x55ed7d379d78) fp(0x55ed7d379de0) pc(0x7f40f06c2ef0) *dart:async__Future@4048458__propagateToListeners@4048458_handleValueCallback ]
[bare-dart : sp(0x55ed7d379df0) fp(0x55ed7d379e30) pc(0x7f40f06c2978) *dart:async__Future@4048458__propagateToListeners@4048458 ]
[bare-dart : sp(0x55ed7d379e40) fp(0x55ed7d379e58) pc(0x7f40f06c3da0) *dart:async__Future@4048458__completeWithValue@4048458 ]
[bare-dart : sp(0x55ed7d379e68) fp(0x55ed7d379e88) pc(0x7f40f06c3398) *dart:async__Future@4048458__asyncComplete@4048458_<anonymous closure> ]
[bare-dart : sp(0x55ed7d379e98) fp(0x55ed7d379ea0) pc(0x7f40f05e60d8) *dart:async_::__microtaskLoop@4048458 ]
[bare-dart : sp(0x55ed7d379eb0) fp(0x55ed7d379ee0) pc(0x7f40f05e735c) *dart:async_::__startMicrotaskLoop@4048458 ]
[bare-dart : sp(0x55ed7d379ef0) fp(0x55ed7d379ef0) pc(0x7f40f05e7478) *dart:async_::__startMicrotaskLoop@4048458__startMicrotaskLoop@4048458 ]
[bare-dart : sp(0x55ed7d379f00) fp(0x55ed7d379f08) pc(0x7f40f0689288) *dart:isolate_::__runPendingImmediateCallback@1026248 ]
[bare-dart : sp(0x55ed7d379f18) fp(0x55ed7d379f18) pc(0x7f40f0601950) *dart:isolate__RawReceivePortImpl@1026248__handleMessage@1026248 ]
[entry    : sp(0x55ed7d379f28) fp(0x55ed7d379fe8) pc(0x7f40f05def68)]

--- Re-run this test:
python tools/test.py -n dartkp-linux-release-simarm64 vm/dart/v8_snapshot_profile_writer_test
```

The revert of that commit didn't change this, interestingly enough, they're still crashing on later commits.

Approving these failures for now to get the bot back to red. /cc @sjindel-google

**Comment by sjindel-google** [comment]
The stacktrace seems to suggest an issue with the original CL, relating to IsSubtypeOf.

**Comment by crelier** [comment]
Thanks for filing this issue and for dealing with the red bots after the revert. The test database was not properly restored after the revert and whesse@ is working on that separate issue.
Fix relanded as f7a9017
